### PR TITLE
[RB17P] Add support for Retevis RB17P radio's new features

### PR DESCRIPTION
**Pull request by kk7ds**
The latest RB17P firmware (v1.31) upgrades this model to 128 channels, customizable channels and adds support for channel names.

This patch adds support for these new features.

Fixes #10071

### CODE REVIEW

File: chirp/drivers/retevis_rb17p.py [hunk 1]
```diff
@@ -382,6 +353,12 @@ def get_memory(self, number):
             mem.duplex = int(_mem.rxfreq) > int(_mem.txfreq) and "-" or "+"
             mem.offset = abs(int(_mem.rxfreq) - int(_mem.txfreq)) * 10
 
+        for char in _nam.name:
```
Comment (goldstar611) [inline chirp/drivers/retevis_rb17p.py:356]: Is `_nam.name` not guaranteed to be a string here?

It feels like it'd be more natural to write something like `mem.name = _nam.name.replace("\xFF", "").rstrip()`

Comment (kk7ds) [reply chirp/drivers/retevis_rb17p.py:356]: No, it's a memobj of bitwise chars. You can str(_mem.name) here and the right thing will happen, but it's not a string itself. This pattern is typical for other drivers, for better or worse :)

That said, this is Jim's patch from the mailing list and I'm just proxying here...

Comment (goldstar611) [reply chirp/drivers/retevis_rb17p.py:356]: Thanks for the explanation.

I'm probably showing my ignorance as I post comments here but I'm trying to learn the ins and outs as they show up  :)

Comment (kk7ds) [reply chirp/drivers/retevis_rb17p.py:356]: No worries, glad someone's paying attention :)

**Pull request merged**